packages/eslint-plugin/src/utils/ast-utils.ts

```diff
@@ -27,9 +27,43 @@ function tryReadBaseJson() {
  */
 export function getBarrelEntryPointByImportScope(
```

Comment:
Can we make this simpler?

The `getOneOrNone` seems to be doing a bit more than that since it also returns the result of the function. Additionally, if we are using generics, then they should extend from something other than `any`.

We want to collect all the paths where the alias is a wildcard and matches the import, and then append the delta to the mapped path.

Something like this would work:
```ts
const result = new Set<string>; // set ensures there are no duplicates

Object.entries(paths).forEach((alias, targets) => {
  if (!alias.endsWith('*')) {
    return;
  }
  const strippedAlias = alias.slice(0, -1); // remove asterisk
  if (!importScope.startsWith(strippedAlias)) {
    return;
  }
  const dynamicPart = importScope.slice(strippedAlias.length);
  targets.forEach(target => {
    result.add(target.replace('*', dynamicPart)); // add interpolated value
  })
})

return Array.from(result);
```

I think this is cleaner and more readable, but most importantly - using primitive functions over higher-order functions (map, filter, flat) improves performance significantly.

Reply:
Used your implementation and validated it against the older code. Looks correct. I took liberty to convert `.forEach(` into `for ... of ...` to be in line with removing higher-order functions from the computation.

Reply:
The `.forEach` is not a higher-order function and is slightly faster than `for...of` and comparable with plain `for`. But this is not a deal breaker so let's just leave it.